### ISSUE TITLE
Fix Linux Support

### ISSUE THREAD

**Issue by VermiumSifell** [ShareX/ShareX#6625]
You heard it right..... Make ShareX great again

**Comment by lawrence-laz** [comment]
"Fix"? Was ShareX ever running natively on Linux?

**Comment by florian-sabonchi** [comment]
> ight..... Make ShareX great aga

You can use ScreenX with Wine or alternatively get it from the Steam Store and use it with Proton.  Alternatively I can recommend Flameshot

**Comment by relief-melone** [comment]
I guess this would probably be a pretty tough one but I would love to see ShareX on Linux. Have not yet found anything on linux that came even close. Flameshot is OK but not as convenient in my opinion

**Comment by VermiumSifell** [comment]
> I guess this would probably be a pretty tough one but I would love to see ShareX on Linux. Have not yet found anything on linux that came even close. Flameshot is OK but not as convenient in my opinion

I think that too, since it's coded in programming languages that's only available on Windows I think.

**Comment by Mastergamer433** [comment]
> > I guess this would probably be a pretty tough one but I would love to see ShareX on Linux. Have not yet found anything on linux that came even close. Flameshot is OK but not as convenient in my opinion
> 
> I think that too, since it's coded in programming languages that's only available on Windows I think.

No, the .net framework works on windows, mac and linux, but sharex uses windows forms, which is like a gtk but windows thing, that cant be run on linux.

**Comment by teauxfu** [comment]
.NET MAUI would be an option too, but it also lacks Linux GUI support for the time being. 

Porting this project to an ASP.NET Core server with an Electron front end seems to be the best option for the time being, but that sounds like a ton of work and trying to do all the image editing etc within browser UI may be a royal pain.

**Comment by lawrence-laz** [comment]
There are already multiple good choices for a cross-platform GUI framework experience that support Linux:

- [Uno platform](https://github.com/unoplatform/uno)
- [Avalonia](https://github.com/AvaloniaUI/Avalonia)
- [ImGui.NET](https://github.com/mellinoe/ImGui.NET) (.NET bindings for the great Dear ImGui)

**Comment by Th3-A6add0n** [comment]
https://github.com/ShareX/ShareX/issues/6110#issuecomment-1556170041

**Comment by Damians0** [comment]
We hope to see ShareX on Linux.

**Comment by m00nk** [comment]
+1

**Comment by z80Evgenij** [comment]
+100500!

**Comment by faith-healer** [comment]
+1

**Comment by NWJ9PB** [comment]
Bump

**Comment by VermiumSifell** [comment]
+100000000

**Comment by rafalohaki** [comment]
when

**Comment by NWJ9PB** [comment]
Bump

**Comment by Xiarno** [comment]
+1
Need this. I use it daily on Windows. So much so that it's one of the only application actively preventing me from switching fully to Linux. I use it *that* much.

**Comment by Xiarno** [comment]
> Flameshot works as an alternative, but for sure miss good old ShareX

I tried Flameshot.
It is **_not_** a decent alternative.

**Comment by Koopzington** [comment]
KDE's Spectacle is fairly close but still not quite as good if you know the shortcuts to modify the selection window. It's still missing the "magnifying glass" and upload capabilities though and annotations are only possible when you open it before taking the screenshot. It also doesn't snap to UI components and windows but i'm not sure if that is even possible on Linux.

**Comment by BrycensRanch** [comment]
The work is being done here <https://github.com/BrycensRanch/SnapX>, of course, it's not ready yet. I am *only* posting this for C# developers who'd like to contribute. If you'd like, you can get into a call with me on Discord. My username is `brycensranch`

Happy holidays, everyone!

**Comment by JdotCarver** [comment]
The only reason why I am not migrating to Linux is because awesome software like ShareX are not available there.

We have all seen the direction Microsoft is taking Windows. _(CopilotAI, Windows Recall, Increased Telemetry, Forcing binding Accounts)_
The more Linux grows, the more people have a healthy alternative, the less aggregious the other OSes will be.

Having ShareX on Linux would benefit everyone.

**Comment by dotSomething** [comment]
+1

**Comment by MicroDevX** [comment]
> > هذا..... اجعل ShareX رائعا
> 
> يمكنك استخدام ScreenX مع Wine أو بدلا من ذلك الحصول عليه من متجر Steam واستخدامه مع Proton. بدلا من ذلك يمكنني أن أوصي ب Flameshot

All of that does not work with Wayland, not even XWayland.

**Comment by MicroDevX** [comment]
+1

**Comment by ItsMaster** [comment]
We still waiting on this one.

**Comment by edisondotme** [comment]
> We still waiting on this one.

It's never going to happen. It's just not.

**Comment by JdotCarver** [comment]
SnapX is doing some promising work though!

**Comment by Edinho367** [comment]
+1

Specially after what MS did to Win10, expect some users to migrate to Linux

They'll search for their favorite softwares in that ecosystem

**Comment by BrycensRanch** [comment]
> +1
> 
> Specially after what MS did to Win10, expect some users to migrate to Linux
> 
> They'll search for their favorite softwares in that ecosystem

Please, no more useless +1 comments... Please take some time to look at my comment.

**Comment by Polda18** [comment]
One thing I really like about ShareX is the support of custom uploaders. So far, no tool on Linux I've been able to find, supports custom uploaders like ShareX does. And because I actually use one myself, I'm still looking for a ShareX Linux alternative that would support custom uploaders. I don't want to use Wine, nor do I want to use Proton. One thing I could do is porting ShareX to Linux myself, tho it requires rewriting the entire code in C++ and change all Windows hooks to Linux hooks instead, possibly with the use of Qt library. Currently, ShareX is using .NET library and is written in C#. That's a huge task and I sadly don't have the necessary time to do it in timely manner. It would take me at least a year to do that alone. Just rewriting the code base. Another huge task would be to keep it up to date with the original, and rewriting each new change into C++ as well. Publishing a package to official popular Linux distros package managers is also quite a daunting task if you ask me. I've never done such a thing.

**Comment by m00nk** [comment]
@Polda18 
I guess there is a way to make this application runs on [mono](https://en.wikipedia.org/wiki/Mono_(software)) under linux. It this case there is no necessary to rewrite all code on another language.

**Comment by Polda18** [comment]
@m00nk That's a possibility. I didn't know that such a feature even exists. And it's currently developed by Microsoft, which is neat, cuz that actually ensures that most of the features of .NET are natively supported. If that compiles to the elf binary native to Linux, then that's step one already in works. The next step is getting it in majority of large Linux distros package managers (Debian's apt -> Ubuntu, Mint; RedHat's dnf -> Fedora, CentOS, AlmaLinux; aur and perhaps then pacman of Arch Linux and its derivatives, flatpak perhaps).

**Comment by BrycensRanch** [comment]
> [@Polda18](https://github.com/Polda18) I guess there is a way to make this application runs on [mono](https://en.wikipedia.org/wiki/Mono_(software)) under linux. It this case there is no necessary to rewrite all code on another language.

No need to change the coding language. C# is cross-platform. When people tried to port ShareX to Mono, it didn't work because the reliance on Windows APIs is strong.  Instead, you must simply rewrite the UI and API calls to be cross-platform. Which is what I have done at <https://github.com/SnapXL/SnapX>, it's still a work in progress, currently basic, but that's how all things start. Try the nightly builds. I haven't made a new release in a while.

**Comment by Polda18** [comment]
> > [@Polda18](https://github.com/Polda18) I guess there is a way to make this application runs on [mono](https://en.wikipedia.org/wiki/Mono_(software)) under linux. It this case there is no necessary to rewrite all code on another language.
> 
> No need to change the coding language. C# is cross-platform. When people tried to port ShareX to Mono, it didn't work because the reliance on Windows APIs is strong. Instead, you must simply rewrite the UI and API calls to be cross-platform. Which is what I have done at https://github.com/SnapXL/SnapX, it's still a work in progress, currently basic, but that's how all things start. Try the nightly builds. I haven't made a new release in a while.

Then I don't need anything to do, maybe just try to find bugs in the code and report them, perhaps try to fix them myself as well, and post pull requests. I guess SnapX is gonna be better alternative for Windows and Linux for me to use. Currently, I use the original ShareX on Windows 11 Canary build (Insider Preview) on my primary (gaming) laptop, while I use the default built-in screenshot tool in Linux Mint on my secondary laptop (used while travelling). How much of a big difference is SnapX opposed to ShareX? Is it pretty much one on one copy, just cross-platform? I suppose it supports custom uploaders just like ShareX, right?

**Comment by BrycensRanch** [comment]
> > > [@Polda18](https://github.com/Polda18) I guess there is a way to make this application runs on [mono](https://en.wikipedia.org/wiki/Mono_(software)) under linux. It this case there is no necessary to rewrite all code on another language.
> > 
> > 
> > No need to change the coding language. C# is cross-platform. When people tried to port ShareX to Mono, it didn't work because the reliance on Windows APIs is strong. Instead, you must simply rewrite the UI and API calls to be cross-platform. Which is what I have done at https://github.com/SnapXL/SnapX, it's still a work in progress, currently basic, but that's how all things start. Try the nightly builds. I haven't made a new release in a while.
> 
> Then I don't need anything to do, maybe just try to find bugs in the code and report them, perhaps try to fix them myself as well, and post pull requests. I guess SnapX is gonna be better alternative for Windows and Linux for me to use. Currently, I use the original ShareX on Windows 11 Canary build (Insider Preview) on my primary (gaming) laptop, while I use the default built-in screenshot tool in Linux Mint on my secondary laptop (used while travelling). How much of a big difference is SnapX opposed to ShareX? Is it pretty much one on one copy, just cross-platform? I suppose it supports custom uploaders just like ShareX, right?

It supports custom uploaders, yeah. It's missing a settings page, so you have to configure it via the JSON files (which are essentially the same as ShareX's)

It has the same goals as ShareX. Just consider it a port that still needs a lot of work before people can daily drive it. Contributors are still much appreciated!

**Comment by Polda18** [comment]
> > > > [@Polda18](https://github.com/Polda18) I guess there is a way to make this application runs on [mono](https://en.wikipedia.org/wiki/Mono_(software)) under linux. It this case there is no necessary to rewrite all code on another language.
> > > 
> > > 
> > > No need to change the coding language. C# is cross-platform. When people tried to port ShareX to Mono, it didn't work because the reliance on Windows APIs is strong. Instead, you must simply rewrite the UI and API calls to be cross-platform. Which is what I have done at https://github.com/SnapXL/SnapX, it's still a work in progress, currently basic, but that's how all things start. Try the nightly builds. I haven't made a new release in a while.
> > 
> > 
> > Then I don't need anything to do, maybe just try to find bugs in the code and report them, perhaps try to fix them myself as well, and post pull requests. I guess SnapX is gonna be better alternative for Windows and Linux for me to use. Currently, I use the original ShareX on Windows 11 Canary build (Insider Preview) on my primary (gaming) laptop, while I use the default built-in screenshot tool in Linux Mint on my secondary laptop (used while travelling). How much of a big difference is SnapX opposed to ShareX? Is it pretty much one on one copy, just cross-platform? I suppose it supports custom uploaders just like ShareX, right?
> 
> It supports custom uploaders, yeah. It's missing a settings page, so you have to configure it via the JSON files (which are essentially the same as ShareX's)
> 
> It has the same goals as ShareX. Just consider it a port that still needs a lot of work before people can daily drive it. Contributors are still much appreciated!

Yeah, I guess I'm not going to use it on my active Windows bare metal machine (my primary, gaming laptop), maybe a VM with Windows 11 inside (where I do not even need to use customizations, so I don't even have to register and activate Windows in it, I still should be able to install software in it tho). Where do I find the nightly builds? Do I compile them myself, or is there any mirror where I can see them? My Linux Mint secondary laptop (used for travelling) uses Cinnamon desktop environment, therefore I'm gonna be maybe the first one to test it under that environment. Otherwise, I've got Ubuntu VMs, so I can test it there.

**Comment by BrycensRanch** [comment]
> > > > > [@Polda18](https://github.com/Polda18) I guess there is a way to make this application runs on [mono](https://en.wikipedia.org/wiki/Mono_(software)) under linux. It this case there is no necessary to rewrite all code on another language.
> > > > 
> > > > 
> > > > No need to change the coding language. C# is cross-platform. When people tried to port ShareX to Mono, it didn't work because the reliance on Windows APIs is strong. Instead, you must simply rewrite the UI and API calls to be cross-platform. Which is what I have done at https://github.com/SnapXL/SnapX, it's still a work in progress, currently basic, but that's how all things start. Try the nightly builds. I haven't made a new release in a while.
> > > 
> > > 
> > > Then I don't need anything to do, maybe just try to find bugs in the code and report them, perhaps try to fix them myself as well, and post pull requests. I guess SnapX is gonna be better alternative for Windows and Linux for me to use. Currently, I use the original ShareX on Windows 11 Canary build (Insider Preview) on my primary (gaming) laptop, while I use the default built-in screenshot tool in Linux Mint on my secondary laptop (used while travelling). How much of a big difference is SnapX opposed to ShareX? Is it pretty much one on one copy, just cross-platform? I suppose it supports custom uploaders just like ShareX, right?
> > 
> > 
> > It supports custom uploaders, yeah. It's missing a settings page, so you have to configure it via the JSON files (which are essentially the same as ShareX's)
> > It has the same goals as ShareX. Just consider it a port that still needs a lot of work before people can daily drive it. Contributors are still much appreciated!
> 
> Yeah, I guess I'm not going to use it on my active Windows bare metal machine (my primary, gaming laptop), maybe a VM with Windows 11 inside (where I do not even need to use customizations, so I don't even have to register and activate Windows in it, I still should be able to install software in it tho). Where do I find the nightly builds? Do I compile them myself, or is there any mirror where I can see them? My Linux Mint secondary laptop (used for travelling) uses Cinnamon desktop environment, therefore I'm gonna be maybe the first one to test it under that environment. Otherwise, I've got Ubuntu VMs, so I can test it there.

https://nightly.link/SnapXL/SnapX/workflows/build/develop?preview

**Comment by ikkqqstudio** [comment]
+1

**Comment by Celtic-Bytes** [comment]
Any update on this?
I moved from windows 11 to linux mint for privacy reasons and it was heartbraking to findout that shareX is not on Linux. I was totally astonished.

**Comment by defaultUser822** [comment]
Gradia is good but it isn't as good though.

**Comment by Gamrin** [comment]
I'd like to ask my voice to this request. Please package this for NixOS. It's literally the best piece of software I don't have access to on Linux.

**Comment by da3dsoul1** [comment]
Linux sucks, hence sharex will never be released there

**Comment by ikkqqstudio** [comment]
> Linux sucks, hence sharex will never be released there

Bro, Linux Is Open-source Spyware-free, Windows And macOS Is Propietary Spyware

**Comment by NoPlagiarism** [comment]
Don't see any mention of XerahS, reimagining of the ShareX user interface built with Avalonia. And it tries to go cross-platform too.

https://github.com/ShareX/XerahS

**Comment by ggtylerr** [comment]
> Don't see any mention of XerahS, reimagining of the ShareX user interface built with Avalonia. And it tries to go cross-platform too.
> 
> https://github.com/ShareX/XerahS

Heavy use of AI :/

EDIT: Also when testing I can't even get it to launch lmao. Couple that with how AI usage is only mentioned on the repo readme and *not* on the website, I can't recommend.
